### PR TITLE
Upgrade to GHC 9.4.8

### PR DESCRIPTION
**Pull request by christiaanb**

### CODE REVIEW

**Review by martijnbastiaan — APPROVED** [review]

**Comment by martijnbastiaan** [comment]
@christiaanb How did you fix the `zlib` issue?

**Comment by DigitalBrains1** [comment]
Drive-by comment: why not GHC 9.6?

**Comment by martijnbastiaan** [comment]
I think the Cabal issue we saw earlier prevented us from upgrading one of the dependencies.

**Comment by christiaanb** [comment]
> @christiaanb How did you fix the `zlib` issue?

It was a cabal.project.freeze issue, in the original PR it disabled the pkgconfig flag for the zlib Haskell package; I simply put it back into the state it was previously: enabled.

File: bittide-experiments/src/Bittide/Topology.hs [hunk 1]
```diff
@@ -256,52 +256,55 @@ instance FromJSON a => FromJSON (TopologyType IO a) where
    where
     tmm = typeMismatch "Topology" v
 
+newtype FUN a = FUN (forall n . SNat n -> a)
```
Comment (kleinreact) [inline bittide-experiments/src/Bittide/Topology.hs:259]: That's an interesting new requirement. Also means we don't need `ImpredicativeTypes` any more.

**Review by kleinreact — COMMENTED** [review]
LGTM

File: bittide-experiments/src/Bittide/Topology.hs (hunk 1 above)
Comment (christiaanb) [reply bittide-experiments/src/Bittide/Topology.hs:259]: GHC was no longer happy with `Maybe (forall n . SNat n -> a)`

**Comment by martijnbastiaan** [comment]
@DigitalBrains1 I looked it up, it was indeed the GHC/Cabal issue:

https://github.com/clash-lang/clash-vexriscv/pull/15:

> I've left 9.6 out, as it is affected by the issues reported in https://github.com/clash-lang/clash-compiler/pull/2665.

**Comment by DigitalBrains1** [comment]
Thanks! That's a good thing to have documented here.

**Pull request merged**